inc/Remote/ApiCore.php

```diff
@@ -582,6 +586,76 @@ public function appendPage($id, $text, $params = array())
         return $this->putPage($id, $currentpage . $text, $params);
     }
 
+    /**
+     * Create one or more users
+     *
+     * @param array[] $userStruct User struct
+     *
+     * @return boolean Create state
+     *
+     * @throws AccessDeniedException
+     * @throws RemoteException
+     */
+    public function createUser($userStruct)
+    {
+        if (!auth_isadmin()) {
+            throw new AccessDeniedException('Only admins are allowed to create users', 114);
+        }
+
+        /** @var \dokuwiki\Extension\AuthPlugin $auth */
+        global $auth;
+
+        if(!$auth->canDo('addUser')) {
+            throw new AccessDeniedException(
+                sprintf('Authentication backend %s can\'t do addUser', $auth->getPluginName()),
+                114
+            );
+        }
+
+        $user = trim($auth->cleanUser($userStruct['user'] ?? ''));
+        $password = $userStruct['password'] ?? '';
+        $name = trim(preg_replace('/[\x00-\x1f:<>&%,;]+/', '', $userStruct['name'] ?? ''));
+        $mail = trim(preg_replace('/[\x00-\x1f:<>&%,;]+/', '', $userStruct['mail'] ?? ''));
+        $groups = $userStruct['groups'] ?? [];
+
+        $notify = (boolean)$userStruct['notify'] ?? false;
+
+        if (empty($user)) {
+            throw new RemoteException('user not supplied', 114);
+        }
+
+        if (empty($name)) {
+            throw new RemoteException('name not supplied', 114);
+        }
+
+        if (!mail_isvalid($mail)) {
+            throw new RemoteException('mail not valid', 114);
+        }
+
+        if(strlen($password) === 0) {
+            $password = auth_pwgen($user);
+        }
+
+        if (!is_array($groups) || count($groups) === 0) {
+            $groups = null;
+        }
+
```

Comment:
Question: the checks above - aren't they all done elsewhere? Eg. in triggerUserMod? I would prefer to rely on existing checks here instead of duplicating checks.

Reply:
TriggerUserMod is more a wrapper with a event trigger, that calls default the `createUser()` function of the auth plugin.

Reply:
This requires also refactoring of the code, similar discussion as with the cleanup above. At the moment the caller of triggerUserMod is responsible. So not in scope for now. Could be filed as issue in the tracker.